### PR TITLE
Added Elasticsearch cluster settings get and put methods plus the flu…

### PR DESCRIPTION
**Pull request by Oloremo**
### What does this PR do?
Adding `get_settings` and `put_settings` cluster method for Elasticsearch module plus the `flush_synced` method.

### What issues does this PR fix or reference?
#54505

### Previous Behavior
Elasticsearch module doesn't support these methods

### New Behavior
Elasticsearch module does support these methods

### Tests written?

Yes

### Commits signed with GPG?

Yes

### CODE REVIEW

**Comment by dwoz** [comment]
re-run full debian10-py3
re-run full centos7-py2

**Review by dwoz — APPROVED** [review]

File: salt/modules/elasticsearch.py [hunk 1]
```diff
@@ -264,6 +264,59 @@ def cluster_stats(nodes=None, hosts=None, profile=None):
         raise CommandExecutionError("Cannot retrieve cluster stats, server returned code {0} with message {1}".format(e.status_code, e.error))
 
 
+def cluster_get_settings(flat_settings=False, include_defaults=False, hosts=None, profile=None):
+    '''
+    .. versionadded:: Sodium
```
Comment (Ch3LL) [inline salt/modules/elasticsearch.py]:
```suggestion
    .. versionadded:: Neon
```
this will be in neon

File: salt/modules/elasticsearch.py [hunk 2]
```diff
@@ -264,6 +264,59 @@ def cluster_stats(nodes=None, hosts=None, profile=None):
...
+
+    try:
+        return es.cluster.get_settings(flat_settings=flat_settings, include_defaults=include_defaults)
+    except elasticsearch.TransportError as e:
+        raise CommandExecutionError("Cannot retrieve cluster settings, server returned code {0} with message {1}".format(e.status_code, e.error))
+
+
+def cluster_put_settings(body=None, flat_settings=False, hosts=None, profile=None):
+    '''
+    .. versionadded:: Sodium
```
Comment (Ch3LL) [inline salt/modules/elasticsearch.py]:
```suggestion
    .. versionadded:: Neon
```

File: salt/modules/elasticsearch.py [hunk 3]
```diff
@@ -1214,3 +1267,44 @@ def snapshot_delete(repository, snapshot, hosts=None, profile=None):
         return True
     except elasticsearch.TransportError as e:
         raise CommandExecutionError("Cannot delete snapshot {0} from repository {1}, server returned code {2} with message {3}".format(snapshot, repository, e.status_code, e.error))
+
+
+def flush_synced(hosts=None, profile=None, **kwargs):
+    '''
+    .. versionadded:: Sodium
```
Comment (Ch3LL) [inline salt/modules/elasticsearch.py]:
```suggestion
    .. versionadded:: Neon
```

**Review by Ch3LL — CHANGES_REQUESTED** [review]
also would you mind add an entry in the CHANGELOG.md file under `Added` for the neon release. thanks

File: salt/modules/elasticsearch.py (hunk 1 above)
Comment (Oloremo) [reply salt/modules/elasticsearch.py]: done

File: salt/modules/elasticsearch.py (hunk 2 above)
Comment (Oloremo) [reply salt/modules/elasticsearch.py]: done

File: salt/modules/elasticsearch.py (hunk 3 above)
Comment (Oloremo) [reply salt/modules/elasticsearch.py]: done

**Comment by Ch3LL** [comment]
@Oloremo not sure if you saw my comment about adding a note to the changelog. This new feature is great and i think would be great to highlight for the neone release. See above for details

**Comment by Oloremo** [comment]
@Ch3LL sorry, missed that. Added.

**Review by Ch3LL — APPROVED** [review]

**Review by s0undt3ch — APPROVED** [review]

**Pull request merged**